multik-api/src/commonMain/kotlin/org/jetbrains/kotlinx/multik/ndarray/data/DataType.kt

```diff
@@ -58,27 +57,30 @@ public enum class DataType(public val nativeCode: Int, public val itemSize: Int,
                 6 -> DoubleDataType
                 7 -> ComplexFloatDataType
                 8 -> ComplexDoubleDataType
-                else -> throw IllegalStateException("One of the primitive types was expected")
+                else -> throw IllegalStateException("One of the primitive types indexes was expected, got $i")
             }
         }
 
         /**
          * Returns [DataType] by class of [element].
          */
-        public fun <T> of(element: T): DataType {
-            return when (element) {
-                is Byte -> ByteDataType
-                is Short -> ShortDataType
-                is Int -> IntDataType
-                is Long -> LongDataType
-                is Float -> FloatDataType
-                is Double -> DoubleDataType
-                is ComplexFloat -> ComplexFloatDataType
-                is ComplexDouble -> ComplexDoubleDataType
-                else -> throw IllegalStateException("One of the primitive types was expected")
+        public inline fun <reified T> of(element: T): DataType {
```

Comment:
This function duplicates `ofKClass`.
Can't we do instance-checks?

Reply:
Sometimes the instance checks don't work on JS https://github.com/Kotlin/multik/pull/20#issuecomment-999227811